### DISCUSSION TITLE
Cloning repositories

### DISCUSSION THREAD

**Discussion by C0d1ngd3v3l0pm3nt** [community/community#34762] [Codespaces]
Hello everyone, I love the community but I have a question to ask.
How do you resolve error 400 when working to clone a repository?

I would appreciate any help on this because I have been stuck for a whole day

**Comment by antonblr** [comment]
Hi @C0d1ngd3v3l0pm3nt - it could help if you provide more info: steps, commands, detailed output.

**Reply by C0d1ngd3v3l0pm3nt** [reply]
I am stuck at git push actually. I enter the command and the message is to enter a password for my Personal access token. I am confused as to what I should do. Does that shed some light on my situation?

**Reply by CaedenPH** [reply]
> I am stuck at git push actually. I enter the command and the message is to enter a password for my Personal access token. I am confused as to what I should do. Does that shed some light on my situation?

Personally, I think that the easiest way to handle this to install `gh` and let that handle your authentication.

**1)** First, install the `gh` binary release; https://github.com/cli/cli/releases/download/v2.16.1/gh_2.16.1_windows_amd64.msi
   (assuming you are on windows)
**2)** Run `gh auth login` in terminal, choose your preferred protocol, probably `https`, then login with the browser and you will be all sorted.

**Reply by airtower-luna** [reply]
@C0d1ngd3v3l0pm3nt, I'm afraid that adds more confusion. Clone and push are very different operations. It's best if you post the commands you use and their output *exactly and in full* (you can censor private repository names, etc.). Otherwise it's hard to say what might be wrong.
